File: Backend_Code/Ingredient.java
package snapchef;

/**
 * Ingredient Class Definition
 *
 * @author Gong-Fan Bao
 * @version 2018-11-14
 */
public class Ingredient {
	private String ingredientName = "";
	private int amount = -1;
	private String unit = "";
	
	//Constructor
	public Ingredient (String ingredientName, int amount, String unit) {
		this.ingredientName = ingredientName;
		this.amount = amount;
		this.unit = unit;
	}
	
	//Main
	public static void main(final String[] args) {
		//Testing
		String iName = "peanut";
		int iAmount = 25;
		String iUnit = "grams";
		
		Ingredient peanut = new Ingredient(iName, iAmount, iUnit);
		
	}
	
	//Getters
	public String getIngredientName() {
		return this.ingredientName;
	}
	
	public int getAmount() {
		return this.amount;
	}
	
	public String getUnit() {
		return this.unit;
	}
	
	//Setters
	public void setIngredientName(String ingredientName) {
		this.ingredientName = ingredientName;
		return;
	}
	
	public void setAmount(int amount) {
		this.amount = amount;
		return;
	}
	
	public void setUnit(String unit) {
		this.unit = unit;
		return;
	}
}

File: Backend_Code/Lambda/Ingredient.py
'''
Created on Dec. 11, 2018

@author: Brandon Benoit
@version: 2018-12-11
'''


class Ingredient (object):
    ingredientName = ""
    quantity = 0
    measureUnit = ""


    def __init__(self, ingredientName,quantity,measureUnit):
        '''
        Constructor
        '''
        Ingredient.ingredientName = ingredientName
        Ingredient.quantity = quantity
        Ingredient.measureUnit = measureUnit
    '''
    Getters
    '''
    def getIngredientName(self) :
        return Ingredient.ingredientName
    
    
    def getAmount(self):
        return Ingredient.amount
    
    
    def getUnit(self):
        return Ingredient.unit
    
    '''
    Setters
    '''
    def setIngredientName(self, ingredientName): 
        Ingredient.ingredientName = ingredientName
        return
    
    
    def setAmount(self, amount) :
        Ingredient.amount = amount
        return
    
    
    def setUnit(self, unit) :
        Ingredient.unit = unit
        return
    
    
    
'''
Main
'''

testIngredient = "Garlic"
testQuantity = 7
testMeasure = "Grams"
try:
    

    peanut =  Ingredient(testIngredient,testMeasure,testQuantity);
    print(peanut.ingredientName,peanut.measureUnit,peanut.quantity)
except AttributeError:
    print ("Not found")
    
    
    
    